modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -4629,3 +4636,175 @@ def from_dataframe(cls, df: "ProtocolDataframe") -> "PandasDataframe":
+        ):
+
+            def case_when(df, name, caselist):  # pragma: no cover
+                caselist = [
+                    tuple(
+                        data.squeeze(1) if isinstance(data, pandas.DataFrame) else data
+                        for data in case_tuple
+                    )
+                    for case_tuple in caselist
+                ]
+                return pandas.DataFrame({name: df.squeeze(1).case_when(caselist)})
+
+            if single_wrap:
+                cls._CASE_WHEN_FN = remote_fn = wrapper_put(case_when)
+            else:
+                remote_fn = case_when
+
+        name = self.columns[0]
+        use_map = single_wrap
+        is_trivial_idx = None
+        # Lists of modin frames: first for conditions, second for replacements
+        modin_lists = [[], []]
+        # Fill values for conditions and replacements respectively
+        fill_values = [True, None]
+        new_caselist = []
+        for case_tuple in caselist:
+            new_case = []
+            for data, modin_list, fill_value in zip(
+                case_tuple, modin_lists, fill_values
+            ):
+                if isinstance(data, cls):
+                    modin_list.append(data)
+                elif callable(data):
+                    if single_wrap:
+                        data = wrapper_put(data)
+                elif isinstance(data, pandas.Series):
+                    use_map = False
+                    if is_trivial_idx is None:
+                        self_idx = self.index
+                        length = len(self_idx)
+                        is_trivial_idx = is_trivial_index(self_idx)
+                    if is_trivial_idx and is_trivial_index(data.index):
+                        data = data[:length]
+                        diff = length - len(data)
+                        if diff > 0:
+                            data = pandas.concat(
+                                [data, pandas.Series([fill_value] * diff)]
+                            )
+                    else:
+                        data = data.reindex(self_idx, fill_value=fill_value)
+                elif use_map and is_list_like(data):
+                    use_map = False
+                new_case.append(data)
+            new_caselist.append(tuple(new_case))
+
+        if modin_lists[0] or modin_lists[1]:
+            # Copartition modin frames
+            use_map = False
+            columns = self.columns
+            column_widths = [1]
+            for modin_list, fill_value in zip(modin_lists, fill_values):
+                _, list_of_right_parts, joined_index, row_lengths = self._copartition(
+                    Axis.ROW_WISE.value,
+                    modin_list,
+                    how="left",
+                    sort=False,
+                    fill_value=fill_value,
+                )
+                modin_list.clear()
+                modin_list.extend(
```

Comment:
Is it okay if we modify the list here? Will `modin_lists` be affected?

Reply:
Yes, this list is used to collect modin frames and then replace them with copartitioned.

Reply:
I mean won't this be a problem for `modin_lists` further or it is done intentionally?

Reply:
Intentionally. This is a temporary list for collecting the modin frames.